logsmanagement/db_api.c

```diff
@@ -0,0 +1,1348 @@
+                            -1, &stmt_insert_log_collection_metadata, NULL);
+    if (unlikely(SQLITE_OK != rc)){
+        throw_error(MAIN_DB, ERR_TYPE_SQLITE, rc, __LINE__, __FILE__, __FUNCTION__);
+        goto return_error;
+    }
+    
+    for (int i = 0; i < p_file_infos_arr->count; i++) {
+
+        struct File_info *const p_file_info = p_file_infos_arr->data[i];
+
+        if(p_file_info->db_mode == LOGS_MANAG_DB_MODE_NONE){
+            p_file_info->db_dir = strdupz("");
+            uv_thread_t *db_writer_thread = mallocz(sizeof(uv_thread_t));
+            rc = uv_thread_create(db_writer_thread, db_writer_db_mode_none, p_file_info);
+            if (unlikely(rc)){
+                throw_error(p_file_info->chart_name, ERR_TYPE_LIBUV, rc, __LINE__, __FILE__, __FUNCTION__);
+                goto return_error;
+            }
+        }
+        else if(p_file_info->db_mode == LOGS_MANAG_DB_MODE_FULL){
+
+            p_file_info->db_mut = mallocz(sizeof(uv_mutex_t));
+            rc = uv_mutex_init(p_file_info->db_mut);
+            if (unlikely(rc)) fatal("Failed to initialize uv_mutex_t");
+            uv_mutex_lock(p_file_info->db_mut);
+
+            // This error check will be used a lot, so define it here. 
+            #define do_sqlite_error_check(p_file_info, rc, rc_expctd){\
+                if(unlikely(rc_expctd != rc)) {\
+                    throw_error(p_file_info->chart_name, ERR_TYPE_SQLITE, rc, __LINE__, __FILE__, __FUNCTION__);\
+                    uv_mutex_unlock(p_file_info->db_mut);\
+                    goto return_error;\
+                }\
+            }
+
+            if(unlikely(
+                    SQLITE_OK != (rc = sqlite3_bind_text(stmt_search_if_log_source_exists, 1, p_file_info->stream_guid, -1, NULL)) ||
+                    SQLITE_OK != (rc = sqlite3_bind_text(stmt_search_if_log_source_exists, 2, p_file_info->filename, -1, NULL)) ||
+                    /* COUNT(*) query should always return SQLITE_ROW */
+                    SQLITE_ROW != (rc = sqlite3_step(stmt_search_if_log_source_exists)))){
+                throw_error(p_file_info->chart_name, ERR_TYPE_SQLITE, rc, __LINE__, __FILE__, __FUNCTION__);
+                uv_mutex_unlock(p_file_info->db_mut);
+                goto return_error;
+            }
+
+            const int log_source_occurences = sqlite3_column_int(stmt_search_if_log_source_exists, 0);
+            switch (log_source_occurences) {
+                case 0: { /* Log collection metadata not found in main DB - create a new record */
+                                        
+                    /* Create directory of collection of logs for the particular 
+                     * log source (in the form of a UUID) and bind it. */
+                    uuid_t uuid;
+                    uuid_generate(uuid);
+                    char uuid_str[GUID_LEN + 1];      // ex. "1b4e28ba-2fa1-11d2-883f-0016d3cca427" + "\0"
```

Comment:
Please, use `UUID_STR_LEN` instead `GUID_LEN +1 `.

Reply:
Done, your suggestion is semantically more correct, although I can see we generally use both.